geode-core/src/main/java/org/apache/geode/internal/cache/InitialImageOperation.java

```diff
@@ -3874,16 +3877,17 @@ private void registerInterestKeys(Map<String, ?> regionKeys, boolean allKey, Loc
             regionsWithInterest.add(regionName);
             if (allKey) {
               ccn.registerClientInterest(regionName, e.getValue(), proxy.getProxyID(), interestType,
-                  isDurable, updatesAsInvalidates, manageEmptyRegions, 0, false);
+                  isDurable, updatesAsInvalidates, manageEmptyRegions, DataPolicy.EMPTY, false);
             } else if (InterestType.REGULAR_EXPRESSION == interestType) {
               for (final Object o : asSet(e.getValue())) {
                 ccn.registerClientInterest(regionName, o, proxy.getProxyID(),
-                    interestType, isDurable, updatesAsInvalidates, manageEmptyRegions, 0, false);
+                    interestType, isDurable, updatesAsInvalidates, manageEmptyRegions,
+                    DataPolicy.EMPTY, false);
               }
             } else {
               ccn.registerClientInterest(regionName, new ArrayList<>(asSet(e.getValue())),
                   proxy.getProxyID(), isDurable, updatesAsInvalidates, manageEmptyRegions,
-                  interestType, false);
+                  DataPolicy.EMPTY, false);
```

Comment:
This seems like a change in behaviour. Previously the value passed as the `regionDataPolicy` argument to `registerClientInterest()` wasn't fixed, as `interestType` could take values 0-4 (although if it was 1 then the `InterestType.REGULAR_EXPRESSION == interestType` check above would prevent us getting to here). Was this just a bug in the old code?

Reply:
@DonalEvans winner winner chicken dinner! You found the bug that validates the need for the PR and the advocation for strong types, like enums, over weaker types, like int. If you look at this method, `registerClientInterest` has overrides that take parameters in slightly different orders but also takes an `int`. In one this `int` was the `InterestType` ordinal and the other it was expected to be, and is later converted to, `DataPolicy` ordinal. You will notice that the compiler was just fine with the bug that that `interestType` was passed for the `dataPolicy` since both were `int`. By converting the parameters to strong types we find this bug at compile time.